### DISCUSSION TITLE
v0.12.10 - The Installer Update

### DISCUSSION THREAD

**Discussion by ThePotatoGuy** [Monika-After-Story/MonikaModDev#9552] [Announcements & Releases]
`[==.......] 22% complete`

# Gameplay Changes

* An installer is now the primary way to install MAS. It can also update MAS, but we recommend to continue using the in-game updater for updates. **The process for manually installing the mod via Mod zips has changed. Use the installer if your system is able to run it.**
    * Links to the OS-specific installers are available in the Assets below. For info on required system specs to run the installer see [this wiki page](https://github.com/Monika-After-Story/mas-installer/wiki/Installer-Required-Specs).
    * Refer to [this wiki](https://github.com/Monika-After-Story/MonikaModDev/wiki/Installation-and-Readme#manual-install) for manual install instructions.

* Several updates to NOU:
    * house rules can now be quickly changed after a game
    * new house rule:
        - reflecting +2 or +4 with a +4 and reflecting skips using a skip of any color

* Monika can now explain the different chess modes to you (new option within the Chess menu).

* Reduced Monika's file size

* Generic sprite pack gifts are now considered good gifts

# Topic Changes

### Additions
* 4 new topics
* 1 new greeting
* 1 new story
* 1 new window react

# Spritepack Changes

### Additions
* Hair Pack 01 by @kkrosie123
* Heart-cut Bikinis by @kkrosie123

# Fixes

* typofixes

* fixed crash when pasting lean sprite codes to EXP Previewer

* fixed crash when a spritepack JSON contains only either `outfit_hair` or `outfit_acs` properties

* fixed glitchy selectors when using `outfit_acs` property

* fixed glitchy selectors when changing clothes/hairs fails

* fixed `monika_lastpoem` being available to everyone

* fixed dialogue render jumpiness when using `extend`

* fixed the "Mind the Gap" being unlocked too early

* fixed issue where Monika wouldn't start a minigame she picked herself in the bored mood

* fixed crash when trying to play NOU with certain submods

* fixed crash when running MAS on certain non-X11 systems

# API Changes

### BREAKING CHANGES

* moved all reset code into python functions - if you previously used a function plugin to hook into `ch30_reset`, be aware that our reset code may run **BEFORE** your code now instead of after.
    * use the `mas_reset.ch30_reset` decorator to mark a function to be called during `ch30_reset`
    * there were some changes to reset execution order - see #8745

### Other changes
* added logging to Window Reacts and catch all x11 exceptions
* new jump with args - `jarg` - see #9413 for more info
* added framework for user-managed API keys. Basically you can include features that require user-supplied API keys now - see #8399 for usage. (wiki page will be added soon)
* `certifi` and `ssl` libs now included - you need to use a special import check to use these - see #8399

### Exp Previewer
* added Neko to available default clothes
* added sunset filter
* spritepack clothes are now sorted alphabetically

## What's Changed
* Add New topic (Again) by @KiRyX112 in https://github.com/Monika-After-Story/MonikaModDev/pull/6863
* [Chess Involve] Simply a in-game chess mode description by @Wingdinggaster656 in https://github.com/Monika-After-Story/MonikaModDev/pull/9456
* Fix `extend` jarring by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9232
* New Topic - Materialism by @Wingdinggaster656 in https://github.com/Monika-After-Story/MonikaModDev/pull/7240
* [contributing] [story] rework of "Knock" story, on #8657 by @my-otter-self in https://github.com/Monika-After-Story/MonikaModDev/pull/9499
* Adds Crunchyroll window react  by @AutumnLeaves991 in https://github.com/Monika-After-Story/MonikaModDev/pull/8305
* Revert "[contributing] [story] rework of "Knock" story, on #8657" by @multimokia in https://github.com/Monika-After-Story/MonikaModDev/pull/9506
* [Pool Topic] - "Do you like pizza?" 🍕 by @lunulae in https://github.com/Monika-After-Story/MonikaModDev/pull/9026
* New love poem greeting by @Erenussocrates in https://github.com/Monika-After-Story/MonikaModDev/pull/7303
* Reinstate PR #9499 by @multimokia in https://github.com/Monika-After-Story/MonikaModDev/pull/9508
* Staring contest Topic by @multimokia in https://github.com/Monika-After-Story/MonikaModDev/pull/9503
* Reduce Monika size by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9440
* EXP previewer fixes by @ThePotatoGuy in https://github.com/Monika-After-Story/MonikaModDev/pull/9505
* [New Spritepack] Heart Cut Bikinis by @kkrosie123 in https://github.com/Monika-After-Story/MonikaModDev/pull/9189
* [New Spritepack] 11 New Hairstyles by @kkrosie123 in https://github.com/Monika-After-Story/MonikaModDev/pull/9202
* outfit JSON props and outfit mode fixes by @ThePotatoGuy in https://github.com/Monika-After-Story/MonikaModDev/pull/9500
* Some more NOU improvements by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9379
* Remove deprecated funcs by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9265
* [Bugfix] - Mind the Gap set to unlock normally by @lunulae in https://github.com/Monika-After-Story/MonikaModDev/pull/8687
* Potentially illogical dialogue fixes by @Wingdinggaster656 in https://github.com/Monika-After-Story/MonikaModDev/pull/9491
* Add API Key settings screen + framework by @ThePotatoGuy in https://github.com/Monika-After-Story/MonikaModDev/pull/8399
* WR cleanup + guardrail by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9480
* Implement augmented jump by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9413
* organized reset code by @ThePotatoGuy in https://github.com/Monika-After-Story/MonikaModDev/pull/8745
* Aff system cleanup by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9040
* Update README.md by @jmwall24 in https://github.com/Monika-After-Story/MonikaModDev/pull/9539
* Typos by @Booplicate in https://github.com/Monika-After-Story/MonikaModDev/pull/9363

## New Contributors
* @KiRyX112 made their first contribution in https://github.com/Monika-After-Story/MonikaModDev/pull/6863
* @my-otter-self made their first contribution in https://github.com/Monika-After-Story/MonikaModDev/pull/9499
* @AutumnLeaves991 made their first contribution in https://github.com/Monika-After-Story/MonikaModDev/pull/8305
* @Erenussocrates made their first contribution in https://github.com/Monika-After-Story/MonikaModDev/pull/7303

**Full Changelog**: https://github.com/Monika-After-Story/MonikaModDev/compare/v0.12.9...v0.12.10

<hr /><em>This discussion was created from the release <a href='https://github.com/Monika-After-Story/MonikaModDev/releases/tag/v0.12.10'>v0.12.10 - The Installer Update</a>.</em>

**Comment by RedAISkye** [comment]
Does the installer work for **Steam** version of DDLC - MAS?

**Reply by ThePotatoGuy** [reply]
When you run the installer, it will prompt you for your DDLC folder, so as long as you know where your DDLC folder is, it will work.

**Reply by RedAISkye** [reply]
> When you run the installer, it will prompt you for your DDLC folder, so as long as you know where your DDLC folder is, it will work.

Alright, I was asking since the in-game updater forces you to manually update if it detects it's a Steam version and not a standalone one.

**Reply by RedAISkye** [reply]
Got the "timeout" error in the middle of the installer downloading so I guess I'll just stick to manual installations since the download is also significantly faster from github via IDM.

**Comment by Capricus02** [comment]
No issues. Just wanted to say that the installer music is a friggen bop lol
